apps/src/lab2/lab2Redux.ts

```diff
@@ -475,7 +475,20 @@ async function setUpAndLoadProject(
     }
   });
   projectManager.addSaveFailListener(() => dispatch(setProjectUpdatedError()));
-  return await projectManager.load();
+  // Figure out if we should reset to start sources. This happens if the url parameter
+  // ?reset=true is present.
+  // This parameter is only used by levelbuilders.
+  const url = new URL(document.URL);
+  const params = new URLSearchParams(url.search);
+  let resetToStartSources = false;
+  if (params.get('reset')) {
+    // Remove the reset parameter from the url so we don't reset again.
+    params.delete('reset');
+    url.search = params.toString();
+    window.history.pushState({}, '', url);
+    resetToStartSources = true;
+  }
```

Comment:
minor: could we use the `queryParams` / `updateQueryParams` in [utils.js](https://github.com/code-dot-org/code-dot-org/blob/93069a4f2c16e1e7771c0deb1394438f08e2b681/apps/src/code-studio/utils.js#L26) here? Looks like `updateQueryParams` has the option to also use `pushState`.

Reply:
that's better, updated!